crates/fuel-core/src/query/message_proof.rs

```diff
@@ -72,65 +174,53 @@ pub async fn message_proof(
     };
 
     // Get the block id from the transaction status if it's ready.
-    let block_id = data
-        .transaction_status(&transaction_id)?
-        .and_then(|status| match status {
-            TransactionStatus::Failed { block_id, .. }
-            | TransactionStatus::Success { block_id, .. } => Some(block_id),
-            TransactionStatus::Submitted { .. }
-            | TransactionStatus::SqueezedOut { .. } => None,
-        });
-
-    // Exit if the status doesn't exist or is not ready.
-    let block_id = match block_id {
-        Some(b) => b,
-        None => return Ok(None),
+    let block_id = match data
+        .transaction_status(&transaction_id)
+        .into_api_result::<TransactionStatus, StorageError>()?
+    {
+        Some(TransactionStatus::Success { block_id, .. }) => block_id,
```

Comment:
I want to highlight that here I changed the business logic, now we don't allow verification for failed transactions.

Reply:
This is probably ok since messages are only set by the VM currently. However there's nothing specifically preventing failed scripts from having a valid message output in the future.